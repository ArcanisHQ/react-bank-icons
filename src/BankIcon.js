import React from 'react';
import { propTypes, defaultProps } from './types';
import { getIconByName, getIconById } from './icon-set';

const BankIcon = ({bankName, bankId, size, color}) => {

    const bankIconData = getIconByName(bankName); 
    const svgPath = (color ? bankIconData.svg.mono : bankIconData.svg.original );

    return (
        <svg width={size} className={'rbbi-' + bankIconData.bankName } height={size} viewBox="0 0 24 24" fill={ color } >
            <title>{bankIconData.ariaTitle}</title>
            <desc>{ '√çcone representando a marca do ' + bankIconData.ariaTitle +'.' }</desc>
            <g dangerouslySetInnerHTML={{__html: svgPath }} />
        </svg>
    )}

BankIcon.propTypes = propTypes;
BankIcon.defaultProps = defaultProps;

export default BankIcon;