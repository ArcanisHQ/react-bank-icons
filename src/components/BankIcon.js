import React from "react";
import { getIconById, getIconByName } from "../util/icon-set";
import FallbackIcon from "./FallbackIcon";
import { defaultProps, propTypes } from "./types/BankIcon.types";

const getIcon = (bankName, bankId) => {
  if (bankName && bankId) {
    return getIconByName(bankName);
  }
  return bankId ? getIconById(bankId) : getIconByName(bankName);
};

const BankIcon = ({
  bankName,
  bankId,
  size,
  viewBox = "0 0 32 32",
  color,
  borderRadius,
  fallbackIcon,
}) => {
  console.log({bankId, bankName});
  const bankIconData = getIcon(bankName, bankId);

  const svgBorderRadius = (size * borderRadius) / 24;

  if (!bankIconData) {
    if (fallbackIcon) {
      return (
        <fallbackIcon size={size} borderRadius={borderRadius} color={color} />
      );
    }
    return (
      <FallbackIcon size={size} borderRadius={borderRadius} color={color} />
    );
  }

  const Svg = bankIconData.svg.original;
  const ariaLabel = `√çcone representando a marca do ${bankIconData.ariaTitle}.`;
  console.log(Svg);
  if (typeof Svg === "function") {
    return (
      <svg
        width={size}
        style={{ borderRadius: svgBorderRadius }}
        className={`rbbi-${bankIconData.bankName}`}
        height={size}
        fill={color}
      >
        <title>{bankIconData.ariaTitle}</title>
        <desc>{ariaLabel}</desc>
        <Svg size={size} svgBorderRadius={svgBorderRadius} />
      </svg>
    );
  }

  return (
    <svg
      width={size}
      style={{ borderRadius: svgBorderRadius }}
      className={`rbbi-${bankIconData.bankName}`}
      height={size}
      viewBox="0 0 24 24"
      fill={color}
    >
      <title>{bankIconData.ariaTitle}</title>
      <desc>{ariaLabel}</desc>
      {Svg}
    </svg>
  );
};

BankIcon.propTypes = propTypes;
BankIcon.defaultProps = defaultProps;

export default BankIcon;
