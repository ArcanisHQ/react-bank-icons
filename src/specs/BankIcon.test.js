jest.mock('../icons');
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import BankIcon from '../BankIcon';

const defaultProps = {
  bankName: 'cajuina',
  bankId: -1,
  size: 16,
};

const BankIconComponent = (props = defaultProps) => shallow(<BankIcon {...props} />);
const findChild = (selector) => BankIconComponent().find(selector);

describe('<BankIcon />', () => {
  it('renders BankIcon component correctly', () => {
    const component = renderer.create(<BankIcon {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should do something', () => {
    expect(true).toBeTruthy();
  });
});
