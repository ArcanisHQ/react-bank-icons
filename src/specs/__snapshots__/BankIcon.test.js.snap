// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankIcon /> renders BankIcon component correctly 1`] = `
<div>
  Hello!
</div>
`;
